modules/userman/oauth.go

```diff
@@ -0,0 +1,281 @@
+		Scopes:       []string{"email"},
+	}
+
+	// state param cookies require HTTPS by default; disable for localhost development
+	//stateConfig := gologin.DebugOnlyCookieConfig
+	stateConfig := gologin.DefaultCookieConfig
+	router.Handle("/v1/api/{project}/auth/{dbType}/oauth/fb", facebook.StateHandler(stateConfig, facebook.LoginHandler(oauth2Config, nil)))
+	router.Handle("/v1/api/{project}/auth/{dbType}/oauth/fb-callback", facebook.StateHandler(stateConfig, facebook.CallbackHandler(oauth2Config, issueFacebookSession(m), nil)))
+}
+
+func issueFacebookSession(m *Module) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		// Get the path parameters
+		vars := mux.Vars(r)
+		project := vars["project"]
+		dbType := vars["dbType"]
+
+		ctx := r.Context()
+		facebookUser, err := facebook.UserFromContext(ctx)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		token, err := m.addUser(ctx, project, dbType, facebookUser.Email, facebookUser.Name)
+		if err != nil {
+			log.Println("Err: ", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+			return
+		}
+
+		// Get the OAuth config
+		c, _ := m.getOAuth("google")
+
+		http.Redirect(w, r, c.RedirectURL+"?token="+token, http.StatusSeeOther)
+	}
+	return http.HandlerFunc(fn)
+}
+
+func (m *Module) newGithubOAuth(router *mux.Router) {
+	// Don't register routes if module is disabled
+	if enabled := m.isActive("github"); !enabled {
+		return
+	}
+
+	// Get the OAuth config
+	c, _ := m.getOAuth("github")
+
+	oauth2Config := &oauth2.Config{
+		ClientID:     c.ID,
+		ClientSecret: c.Secret,
+		RedirectURL:  c.Host + "/v1/oauth/github-callback",
+		Endpoint:     githubOAuth2.Endpoint,
+	}
+	// state param cookies require HTTPS by default; disable for localhost development
+	stateConfig := gologin.DefaultCookieConfig
+	router.Handle("/v1/api/{project}/auth/{dbType}/oauth/github", github.StateHandler(stateConfig, github.LoginHandler(oauth2Config, nil)))
+	router.Handle("/v1/api/{project}/auth/{dbType}/oauth/github-callback", github.StateHandler(stateConfig, github.CallbackHandler(oauth2Config, issueGithubSession(m), nil)))
+}
+
+func issueGithubSession(m *Module) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		// Get the path parameters
+		vars := mux.Vars(r)
+		project := vars["project"]
+		dbType := vars["dbType"]
+
+		ctx := r.Context()
+		githubUser, err := github.UserFromContext(ctx)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		token, err := m.addUser(ctx, project, dbType, githubUser.GetEmail(), githubUser.GetName())
+		if err != nil {
+			log.Println("Err: ", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+			return
+		}
+
+		// Get the OAuth config
+		c, _ := m.getOAuth("google")
```

Comment:
Shouldn't this be `m.getOAuth("github")`?

Reply:
resolved